sqlmesh/core/engine_adapter/athena.py

```diff
@@ -90,14 +97,7 @@ def _get_data_objects(
         schema = schema_name.db
         query = (
             exp.select(
-                exp.case()
-                .when(
-                    # calling code expects data objects in the default catalog to have their catalog set to None
-                    exp.column("table_catalog", table="t").eq("awsdatacatalog"),
-                    exp.Null(),
-                )
-                .else_(exp.column("table_catalog"))
-                .as_("catalog"),
+                exp.column("table_catalog").as_("catalog"),
```

Comment:
Why do we no longer need this?

Reply:
I hit [this problem](https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/dbt/loader.py#L170) which triggered me to analyse catalog support some more.

Athena has the concept of catalogs, just not setting them or querying the current one after a connection is established. So I changed `CATALOG_SUPPORT` to `CatalogSupport.SINGLE_CATALOG_ONLY` and implemented `get_current_catalog()` in terms of reading the catalog off the PyAthena connection.

Since the connection catalog defaults to `awsdatacatalog` when it's not set, now `get_default_catalog()` returns eg `awsdatacatalog` instead of `None` so I needed to change `get_data_objects()` to return `awsdatacatalog` instead of `None` so that code checking if the "current" catalog = the default catalog would continue to work